packages/@sanity/form-builder/src/inputs/arrays/ArrayOfPrimitivesInput/ArrayOfPrimitivesInput.tsx

```diff
@@ -184,10 +266,16 @@ export class ArrayOfPrimitivesInput extends React.PureComponent<Props> {
                     const childPresence = presence.filter((pItem) =>
                       startsWith([index], pItem.path)
                     )
+                    const memberType = this.getMemberType(resolveTypeName(item))
 
+                    // Best effort attempt to make a stable key for each item in the array
+                    // Since items may be reordered and change at any time, there's no way to reliably address each item uniquely
+                    // (hashing the item value would impact performance, and it's common to have duplicate values)
```

Comment:
might be worth deleting this comment line. i don't think you'd want to hash by value because every time you change the value, it'll cause the component to unmount and remount